js/iframe-coordinator/libs/client.ts

```diff
@@ -61,19 +61,23 @@ export default {
   }
 };
 
-function start() {
+function start(frameWindow: Window) {
```

Comment:
This is probably the most controversial thing.

I wasn't allowed to set global.window (its readonly). So if we want to mock window, we have to inject the dependencies.  Now that I think about it more in the review, I think we may want to make this a config object instead of a parameter.  But for now i'll leave this up for comments/alternatives.

Reply:
Yeah, no getting around the need to mock window.  Been there, go the t-shirt ;)

I've definitely done it like this and passed it in with `window` being the default value.

I've also used a `_window` "private" variable before.  I don't make it a  proper `private` but rely on the underscore convention for people not to use it.  Then, I can set the property to a mocked object in my tests and not pollute the API contract with testing concerns.

Both will work fine though.

Reply:
I'm fine with any approach to enable mocking window. My only strong opinion here is that under normal usage, consumers shouldn't have to explicitly pass in the window object.

Reply:
I'll leave this here for now, until i merge in the class changes from the other pr.  That will move this into the constructor and change it up a bit anyways.